runway/cfngin/hooks/aws_lambda.py

```diff
@@ -163,6 +173,195 @@ def _zip_from_file_patterns(root, includes, excludes, follow_symlinks):
+    pip_cmd = (
+        'python -m pip install -t /var/task -r /var/task/requirements.txt'
+    )
+
+    LOGGER.info('lambda.docker: Using docker image "%s" to build deployment '
+                'package...', docker_image)
+
+    service = docker_client.containers.run(image=docker_image,
+                                           command=['/bin/sh', '-c', pip_cmd],
+                                           auto_remove=True,
+                                           detach=True,
+                                           mounts=[work_dir_mount])
+
+    # 'stream' creates a blocking generator that allows for real-time logs.
+    # this loop ends when the container 'auto_remove's itself.
+    for log in service.logs(stdout=True,
+                            stderr=True,
+                            stream=True,
+                            tail=0):
+        # without strip there are a bunch blank lines in the output
+        LOGGER.info('lambda.docker: %s', log.decode().strip())
+
+
+def _zip_package(package_root, includes, excludes, dockerize_pip=False,
+                 follow_symlinks=False, use_pipenv=False, **kwargs):
+    """Create zip file in memory with restored packages.
+
+    Args:
+        package_root (str): Base directory to copy files from.
+        includes (List[str]): Inclusion patterns. Only files  matching those
+            patterns will be included in the result.
+        excludes (List[str]): Exclusion patterns. Files matching those
+            patterns will be excluded from the result. Exclusions take
+            precedence over inclusions.
+        dockerize_pip (Union[bool, str]): Whether to use docker or under what
+            conditions docker will be used to run ``pip``.
+        follow_symlinks (bool): If true, symlinks will be included in the
+            resulting zip file.
+        use_pipenv (bool): Wether to use pipenv to export a Pipfile as
+            requirements.txt.
+        kwargs (Any): Advanced options for subprocess and docker. See source
+            code to determine what is supported.
+
+    Returns:
+        Tuple[str, str]: Content of the ZIP file as a byte string and
+        calculated hash of all the files
+
+    """
+    temp_root = os.path.join(os.path.expanduser('~'), '.runway_cache')
+    if not os.path.isdir(temp_root):
+        os.makedirs(temp_root)
+
+    with tempfile.TemporaryDirectory(
+            prefix='cfngin', dir=temp_root
+    ) as tmpdir:
+        tmp_req = os.path.join(tmpdir, 'requirements.txt')
+        if use_pipenv:
+            _handle_use_pipenv(package_root, tmpdir,
+                               kwargs.get('pipenv_lock_timeout', 300))
+            # dir created when creating a lock file
+            excludes.append('.venv/')
+        for file_name in _find_files(package_root, includes, excludes,
+                                     follow_symlinks):
+            copyfile(os.path.join(package_root, file_name),
```

Comment:
this breaks if the package has any directories. I added `lib/__init__.py` to `dockerize_src` to break this for the example in integration_tests